server/db/driver/pg/internal/matches.go

```diff
@@ -150,4 +150,10 @@ const (
 
 	SetSwapDone = `UPDATE %s SET active = FALSE
 		WHERE matchid = $1;`
+
+	SelectMatchStatuses = `SELECT matchid, status, aContract, bContract, aContractCoinID,
+		bContractCoinID, aRedeemCoinID, bRedeemCoinID, aRedeemSecret
+		FROM %s
+		WHERE matchid IN (%s)
```

Comment:
I've hit the `IN` problem before, but found that `= ANY` is supported out of the box with our pg sql driver .  There still might be issues with MatchID and BYTEA arrays, but it at least handles the syntax better.  This is because `= ANY` works with array syntax, while `IN` uses a comma separated value syntax.  The pq driver deals with bytea arrays naturally in the first way, so these tend to just work. https://stackoverflow.com/questions/31191507/how-to-use-any-instead-of-in-in-a-where-clause-with-rails/31192557#31192557  If you can get it to work with `= ANY`, the lib/pq driver will do the array syntax stuff for us.  But if it works, it works.

Reply:
So in practice, using `ANY`, I would create a `pq.ByteaArray` and pass it in as a placeholder parameter `$2` instead?

Reply:
It would allow you to use a $ parameter, right.  I can't every recall the rules about implict conversions vs the need to write Valuers, but the idea is that the pq driver could handle the array syntax that works with any.  There may well be more hoops to jump through with a MatchID, but with slices of other native types, `=ANY($1)` just works.

Reply:
Great. I was looking for that, but the documentation on `pq.ByteaArray` is pretty thin. Thanks.

Reply:
Might be some examples with dcrdata, can't recall.

Reply:
I kinda think you can use `pq.Array(matchids)` or `pq.GenericArray(matchids)` as long as MatchID has a Valuer that returns a supported type like `[]byte`.

Reply:
I might be missing something, but neither of those seemed to work. I'm seeing `pq: invalid byte sequence for encoding "UTF8": 0xa9`

Reply:
Oh, well.  No biggie.  It looks like that makes it do something like `{"crazyunicodecharacters"}` instead of `{"\xhexencodding"}`.  Not what I would expect, but not worth worrying about.  The ByteArray approach is good.